stonesoup/metricgenerator/tests/test_basicmetrics.py

```diff
@@ -51,11 +51,10 @@ def test_basicmetrics():
                                            end_timestamp=start_time +
                                            datetime.timedelta(seconds=4)),
                                        generator=generator)}
-
     for metric_name in ["Number of targets",
                         "Number of tracks", "Track-to-target ratio"]:
         calc_metric = [i for i in correct_metrics if i.title == metric_name][0]
-        meas_metric = [i for i in metrics if i.title == metric_name][0]
+        meas_metric = [metrics[i] for i in metrics if i == metric_name][0]
```

Comment:
```suggestion
        meas_metric = metrics.get(metric_name)
```

Then check if meas_metric is `None` etc.